src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -228,13 +353,20 @@ public int hashCode()
     private boolean checkForFreeSpace(final List<IItemHandler> handlers)
     {
         final List<ItemStack> secondaryStacks = new ArrayList<>();
-        for (final ItemStack stack : input)
+        if(!secondaryOutputs.isEmpty())
+        {
+            secondaryStacks.addAll(secondaryOutputs);
+        }
+        else
         {
-            final ItemStack container = stack.getItem().getContainerItem(stack);
-            if (!ItemStackUtils.isEmpty(container))
+            for (final ItemStack stack : input)
             {
-                container.setCount(stack.getCount());
-                secondaryStacks.add(container);
+                final ItemStack container = stack.getItem().getContainerItem(stack);
+                if (!ItemStackUtils.isEmpty(container))
+                {
+                    container.setCount(stack.getCount());
```

Comment:
if you're doing modifications in a simple checkForSpace() function you at least have to properly document/comment why that is there/whatfor it is. Or maybe even better rename the function

Reply:
Adding a few comments, renaming a variable, but not the function. It should be more clear now.